tests/cupy_tests/sparse_tests/test_linalg.py

```diff
@@ -0,0 +1,41 @@
+import unittest
+
+import numpy
+import scipy.sparse.linalg  # NOQA
+
+import cupy.sparse.linalg  # NOQA
+from cupy import testing
+
+
+@testing.parameterize(*testing.product({
+    'dtype': [numpy.float32, numpy.float64],
+}))
+@testing.with_requires('scipy')
+class TestLsqr(unittest.TestCase):
+
+    def setUp(self):
+        self.A = numpy.random.randint(2, size=(10, 10))
+        self.b = numpy.random.randint(2, size=10)
+
+    @testing.numpy_cupy_allclose(atol=1e-3, sp_name='sp')
+    def test_csrmatrix(self, xp, sp):
+        A = xp.array(self.A).astype(self.dtype)
+        b = xp.array(self.b).astype(self.dtype)
+        A = A + A.T
+        A = sp.csr_matrix(A)
+        x = sp.linalg.lsqr(A, b)
+        if xp == numpy:
```

Comment:
Why is this branch requied?

Reply:
I changed the statement to `sp == scipy.sparse`. There are two reasons why I inserted it. First, the return value of `scipy.sparse.linalg.lsqr` is a tuple and the first element of the tuple is the answer `x`. So we have to extract it in scipy case. Second, the dtype of the answer `x` by scipy is always float64 even when the input dtype is float32. So we have to match the dtype with cupy one.